src/visualizers/SeqColor.ts

```diff
@@ -0,0 +1,344 @@
+and the color of each circle represents 
+the color combination of its prime factors. \s
+
+Press the "Left arrow key" to show/hide the color of each prime factor.\s
+Press the "Right arrow key" to show/hide labels.\s
+Press "[" to turn up brightness.\s
+Press "]" to turn down brightness. \s
+ **/
+
+const colorMap = new Map()
+
+// Show description box by pressing left arrow key
+class SeqColor extends VisualizerDefault implements VisualizerInterface {
+    name = 'Sequence Color'
+    n = 40
+
+    params = {
+        n: {
+            value: this.n,
+            forceType: 'integer',
+            displayName: 'number of terms',
+            required: true,
+        },
+    }
+
+    private currentNum = 0
+    private X = 0
+    private Y = 0
+    private boxSizeX = 800
+    private boxSizeY = 90
+    //private boxSize = this.sketch.createVector(800, 90)
+    private canvasSizeX = 800
+    private canvasSizeY = 800
+    private subG = this.sketch.createGraphics(this.boxSizeX, this.boxSizeY)
+    private subL = this.sketch.createGraphics(
+        this.canvasSizeX,
+        this.canvasSizeY
+    )
+    private boxIsShow = false
+    private primeNum: number[] = []
+    private countPrime = 0
+    private firstDraw = true
+    private showLabel = false
+    private brightAdjust = 100
+
+    initialize(sketch: p5, seq: SequenceInterface) {
+        super.initialize(sketch, seq)
+        this.sketch = sketch
+        this.seq = seq
+        this.currentNum = seq.first
+        this.X = 0
+        this.Y = 0
+        this.ready = true
+    }
+
+    setup() {
+        this.sketch.background('black')
+        this.sketch.colorMode(this.sketch.HSB, 360, 100, 100)
+        this.sketch.frameRate(30)
+
+        this.firstDraw = true
+
+        // Set position of the circle
+        this.X = this.canvasSizeX / this.n + 30
+        this.Y = this.canvasSizeX / this.n + 50
+
+        // Obtain all prime numbers from the sequence
+        for (let i = this.seq.first; i < this.n; i++) {
+            const checkCurrentPrime = Number(this.seq.getElement(i))
+            if (
+                this.isPrime(checkCurrentPrime)
+                && !this.primeNum.includes(checkCurrentPrime)
+            ) {
+                this.primeNum.push(checkCurrentPrime)
+                this.countPrime += 1
+            }
+        }
+    }
+
+    draw() {
+        if (this.firstDraw == true && this.currentNum < this.n) {
+            const currentElement = Number(
+                this.seq.getElement(this.currentNum++)
+            )
+            this.drawCircle(currentElement)
+
+            this.changePosition()
+
+            // Check if drawing finished
+            if (this.currentNum >= this.n) {
+                this.firstDraw = false
+            }
+            this.sketch.noStroke()
+        } else {
+            // Monitor keyboard events after finishing drawing
+            // Bug: multiple clicks detected when only one click happened
```

Comment:
This bug needs resolving.  I gave it a go, however, it looks like it requires `keyPressed()` from p5.js, and this isn't working; maybe similar to #120 ?

Reply:
Moot since key interaction has been removed. Resolving.